index.js

```diff
@@ -30,26 +43,29 @@ class BindingComponent extends React.Component {
         val = this.props.bindStateType(val);
       }
     }
-    const names = this.props.bindStateName.split('.');
-    if (names.length === 1) {
-      this.props.bindStateCtx.setState({
-        [names[0]]: val
-      });
-    } else {
-      const newState = this.props.bindStateCtx.state[names[0]];
-      const subnames = names.slice(1);
-      subnames.reduce((curr, name, index) => {
-        if (index === subnames.length - 1) {
-          curr[name] = val;
-        } else {
-          curr = curr[name];
-        }
-        return curr;
-      }, newState);
-      this.props.bindStateCtx.setState({
-        [names[0]]: newState
-      });
-    }
+
+    this.props.bindStateCtx.setState((prevState, currentProps) => {
+      let path;
+      let pathReversed;
+      let overlay = val;
+      let pathIndex = 0;
+      try {
+        path = currentProps.bindStateName.split('.');
+        pathReversed = path.reverse();
+      } catch (error) {
+        return prevState;
+      }
+      do {
+        let subpath = pathReversed[pathIndex];
```

Comment:
Or declare subpath at the very beginning of function body, which I think may be better.